### PR TITLE
Add bdb stubs

### PR DESCRIPTION
**Pull request by CraftSpider**
Stubs for the bdb module

### CODE REVIEW

File: stdlib/2and3/bdb.pyi [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+_TraceDispatch = Callable[[FrameType, str, Any], Any]  # TODO: Recursive type
+_ExcInfo = Tuple[Type[BaseException], BaseException, FrameType]
+
+GENERATOR_AND_COROUTINE_FLAGS: int = ...
+
+class BdbQuit(Exception): ...
+
+class Bdb:
+
+    skip: Set[str]
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: Should be Optional

File: stdlib/2and3/bdb.pyi [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    def reset(self) -> None: ...
+    def trace_dispatch(self, frame: FrameType, event: str, arg: Any) -> _TraceDispatch: ...
+    def dispatch_line(self, frame: FrameType) -> _TraceDispatch: ...
+    def dispatch_call(self, frame: FrameType, arg: None) -> _TraceDispatch: ...
+    def dispatch_return(self, frame: FrameType, arg: Any) -> _TraceDispatch: ...
+    def dispatch_exception(self, frame: FrameType, arg: _ExcInfo) -> _TraceDispatch: ...
+    def is_skipped_module(self, module_name: str) -> bool: ...
+    def stop_here(self, frame: FrameType) -> bool: ...
+    def break_here(self, frame: FrameType) -> bool: ...
+    def do_clear(self, arg) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: arg: Any

File: stdlib/2and3/bdb.pyi [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    def break_anywhere(self, frame: FrameType) -> bool: ...
+    def user_call(self, frame: FrameType, argument_list: None) -> None: ...
+    def user_line(self, frame: FrameType) -> None: ...
+    def user_return(self, frame: FrameType, return_value: Any) -> None: ...
+    def user_exception(self, frame: FrameType, exc_info: _ExcInfo) -> None: ...
+    def set_until(self, frame: FrameType, lineno: int = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: lineno is Optional

File: stdlib/2and3/bdb.pyi [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    def set_step(self) -> None: ...
+    def set_next(self, frame: FrameType) -> None: ...
+    def set_return(self, frame: FrameType) -> None: ...
+    def set_trace(self, frame: FrameType = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: Remove `= ...`

File: stdlib/2and3/bdb.pyi [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    def set_continue(self) -> None: ...
+    def set_quit(self) -> None: ...
+    def set_break(self, filename: str, lineno: int, temporary: bool = ..., cond: str = ..., funcname: str = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: cond and funcname are Optional

File: stdlib/2and3/bdb.pyi [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    def get_bpbynumber(self, arg: SupportsInt) -> Breakpoint: ...
+    def get_break(self, filename: str, lineno: int) -> bool: ...
+    def get_breaks(self, filename: str, lineno: int) -> List[Breakpoint]: ...
+    def get_file_breaks(self, filename: str) -> List[Breakpoint]: ...
+    def get_all_breaks(self) -> List[Breakpoint]: ...
+    def get_stack(self, f: FrameType, t: TracebackType) -> Tuple[List[Tuple[FrameType, int]], int]: ...
+    def format_stack_entry(self, frame_lineno: int, lprefix: str = ...) -> str: ...
+    def run(self, cmd: Union[str, CodeType], globals: Dict[str, Any] = ..., locals: Dict[str, Any] = ...) -> None: ...
+    def runeval(self, expr: str, globals: Dict[str, Any] = ..., locals: Dict[str, Any] = ...) -> None: ...
+    def runctx(self, cmd: Union[str, CodeType], globals: Dict[str, Any], locals: Dict[str, Any]) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: globals and locals are Optional for all of these

File: stdlib/2and3/bdb.pyi [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    def runcall(self, func: Callable[[Any], _T], *args: Any, **kwds: Any) -> _T: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: Returns `Optional[_T]`

File: stdlib/2and3/bdb.pyi [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+class Breakpoint:
+
+    next: int = ...
+    bplist: Dict[Tuple[str, int], List[Breakpoint]] = ...
+    bpbynumber: List[Optional[Breakpoint]] = ...
+
+    funcname: str
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: Optional

File: stdlib/2and3/bdb.pyi [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    func_first_executable_line: Optional[int]
+    file: str
+    line: int
+    temporary: bool
+    cond: str
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: Optional

File: stdlib/2and3/bdb.pyi [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    enabled: bool
+    ignore: int
+    hits: int
+    number: int
+
+    def __init__(self, file: str, line: int, temporary: bool = ..., cond: str = ..., funcname: str = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: cond and funcname are Optional

File: stdlib/2and3/bdb.pyi [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    def deleteMe(self) -> None: ...
+    def enable(self) -> None: ...
+    def disable(self) -> None: ...
+    def bpprint(self, out: IO[str] = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/bdb.pyi]: out is Optional

File: stdlib/2and3/bdb.pyi (hunk 4 above)
Comment (CraftSpider) [reply stdlib/2and3/bdb.pyi]: Are you sure? Because the actual signature for both python 2 and 3.6/7 is `def set_trace(self, frame=None)`

Comment (JelleZijlstra) [reply stdlib/2and3/bdb.pyi]: You're right, not sure why I wrote that.

**Pull request merged**